src/basic/user-util.c

```diff
@@ -720,10 +722,18 @@ bool valid_user_group_name_full(const char *u, bool strict) {
 
         if (!(u[0] >= 'a' && u[0] <= 'z') &&
             !(u[0] >= 'A' && u[0] <= 'Z') &&
+            !(u[0] >= '0' && u[0] <= '9') &&
             u[0] != '_')
                 return false;
 
-        bool warned = false;
+        if (u[0] >= '0' && u[0] <= '9') {
+                if (strict)
+                        return false;
+                else {
+                        log_warning("Bad user or group name \"%s\", accepting for compatibility.", u);
+                        warned = true;
+                }
+        }
```

Comment:
As written, this allows names that consist only of digits. Those should be filtered out.

Reply:
Done.